docs/guides/using-azure-artifacts.md

```diff
@@ -0,0 +1,30 @@
+# Using Azure Artifacts
+
+`uv` can install packages from [Azure DevOps Artifacts](https://learn.microsoft.com/en-us/azure/devops/artifacts/start-using-azure-artifacts?view=azure-devops&tabs=nuget%2Cnugetserver). You can authenticate to a feed using a [Personal Access Token](https://learn.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=Windows) or interactively using Keyring.
+## Authenticate using a PAT
+If you have a Personal Access Token available (eg [`$(System.AccessToken)` in an Azure pipeline](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#systemaccesstoken)), you can authenticate to Artifacts using Basic Auth. Simply include the PAT in the password field of the URL. The username can be any dummy string.
+
+```bash
+# if your token is in the ADO_PAT environment variable
+export UV_EXTRA_INDEX_URL=https://dummy:$ADO_PAT@pkgs.dev.azure.com/{organisation}/{project}/_packaging/{feedName}/pypi/simple/
+
+uv pip install my-private-package
+```
+
+## Authenticate using Keyring
+If you don’t have a PAT handy, you can authenticate to Artifacts using [`keyring`](https://github.com/jaraco/keyring) with [the `artifacts-keyring` plugin](https://github.com/Microsoft/artifacts-keyring). The plugin supports a few different authentication modes including interactive login. Because you’ll be using these two packages to authenticate to Azure Artifacts, you should arrange to have them installed into your environment from a source other than Artifacts.
```

Comment:
The [credential provider tool](https://github.com/microsoft/artifacts-credprovider) has a few different modes and I don’t know whether _all_ of them wind up creating a PAT behind the scenes.

I wanted to keep this doc relatively short to help people get up and running, but I could link to the repo for the cred-provider so there’s enough of a paper trail for people to learn more, what do you think?

Reply:
I can confirm that if the following environment variable is set as follows, a PAT is not created behind the scenes:

```
NUGET_CREDENTIALPROVIDER_VSTS_TOKENTYPE=SelfDescribing
```

Here is the relevant help from the credential provider tool (`dotnet exec CredentialProvider.Microsoft.dll -h`):
```
Token Type
    NUGET_CREDENTIALPROVIDER_VSTS_TOKENTYPE
        Specify 'Compact' to generate a Personal Access Token, which may
        have a long validity period as it can easily be revoked from the UI,
        and sends a notification mail on creation.
        Specify 'SelfDescribing' to generate a shorter-lived JWT token,
        which does not appear in any UI or notifications
        and is more difficult to revoke.
        By default PATs are generated rather than JWTs,
        unless authentication can be performed non-interactively.
```

Reply:
Got it. I would prefer to keep this doc as just a high-level getting started guide, I don’t think we should include advanced configuration like that here. I’ve added a link to the cred provider’s GitHub repo so people can learn more if they want to.

Reply:
Looks good to me :+1: